### ISSUE TITLE
RootEntityType Compilation Error when field name conflicts with Operation name

### ISSUE THREAD

**Issue by jriosdev** [apollographql/apollo-ios#3036]
### Summary

`'Data' is not a member type of struct 'SchemaName.QueryName.Data.QueryName.Result'`

while generating code in 1.1.0
` public typealias RootEntityType = QueryName.Result`
after the update generating code in 1.2.0
` public typealias RootEntityType = QueryName.Data.QueryName.Result`

<img width="1014" alt="Screenshot 2023-05-22 at 4 57 42 PM" src="https://github.com/apollographql/apollo-ios/assets/38908155/35001f2e-44c1-4c9a-9a27-74de456cdf32">

<img width="906" alt="Screenshot 2023-05-22 at 4 58 01 PM" src="https://github.com/apollographql/apollo-ios/assets/38908155/4d89da14-7fe9-421c-b7da-ffdc5e998113">

### Version

1.2.0

### Steps to reproduce the behavior

The issue does not exhibit with all models, so far it seems like its only inline union types

code gen config
`{
  "schemaNamespace" : "Schema",
  "input" : {
    "operationSearchPaths" : [
      "**/*.graphql"
    ],
    "schemaSearchPaths" : [
      "**/*.graphqls"
    ]
  },
  "output" : {
    "testMocks" : {
      "none" : {
      }
    },
    "schemaTypes" : {
      "path" : "../../../../Schema",
      "moduleType" : {
        "swiftPackageManager" : {
        }
      }
    },
    "operations" : {
      "inSchemaModule" : {
      }
    }
  }
}
`

**Comment by AnthonyMDev** [comment]
Thanks for the report. I see exactly what's going on here. Your query is named `AirportAndCityAutocompleteQuery`, but it includes a field called `airportAndCityAutocompleteQuery`, which is generating a selection set named `AirportAndCityAutocompleteQuery` inside of the operation definition (which is also defined with the name `AirportAndCityAutocompleteQuery`)! 🤯  It's a query-ception!

We will discuss how we can address this for the future, either by handling the generated naming differently or by throwing a validation error during code generation if we detect this situation. 

In the mean time, the simple workaround for you is to put a field alias on the `airportAndCityAutocompleteQuery` field in your operation.

```graphql
query AirportAndCityAutocomplete($searchString: String!) {
  searchResult: airportAndCityAutocompleteQuery(searchString: $searchString) {
    ...
  }
}
```

Of course the alias `searchResult` can be whatever you want it to be. This alias will change the response data internal struct to be `AirportAndCityAutocompleteQuery.Data.SearchResult` and the field name on the response data object would change from `airportAndCityAutocompleteQuery` to `searchResult`.

**Comment by iAmericanBoy** [comment]
I might be running into the same or something similar on 1.3.0:
 'Analysis' is not a member type of struct 'M1Schema.ApolloPortfolioTopSlice.AsRootPortfolioSlice.To.AsUserPie'
<img width="896" alt="Screenshot 2023-07-05 at 7 02 33 PM" src="https://github.com/apollographql/apollo-ios/assets/32912858/b835fd19-426e-4e80-b9fe-f61fdd0862dd">

we have a query called PortfolioSlice that also has a type alias node:
```graphql
query PortfolioSlice(...) {
  PortfolioSlice: node(...) {
    ... ApolloPortfolioTopSlice
  }
 fragment ApolloPortfolioTopSlice on PortfolioSlice {
  to {
    ...on UserPie {
...
      analysis {
      ...
      }
	}
    ...on SystemPie {
      analysis {
      ...
      }
    }
  }
} 
...

  ```
but after changing that node type alias name it is still happening so im not sure if its not related. is it maybe because we have two fields called `analysis` in that fragment? Happy to share more of schema/query privately

**Comment by AnthonyMDev** [comment]
@iAmericanBoy, thanks for the additional example. I'm not sure what's going on in your case exactly. If you are able to share a schema/query with me you can send it to anthony@apollographql.com. That would help out immensely.

**Comment by iAmericanBoy** [comment]
Thank you! I’ll send it tomorrow morning! I was able to move on from the error by commenting out both analysis sections from the query. So there’s definitely something wrong with how we are doing this.

**Comment by iAmericanBoy** [comment]
@AnthonyMDev Do you have any updates on this? I know its summer vacation everywhere else too, so I just wanted to see if you need anything else from me to get this reproduced.

**Comment by AnthonyMDev** [comment]
@iAmericanBoy, the issue you highlighted [here](https://github.com/apollographql/apollo-ios/issues/3036#issuecomment-1622658245) is not related to the name conflict that this issue is about. However I have figured out the underlying cause for this one and got a fix up for it in #3168!

**Comment by bignimbus** [comment]
Closing this as resolved via #3168 🚀
